src/App.tsx

```diff
@@ -10,6 +10,11 @@ import { AgentContext, AgentProvider } from './context/AgentContext';
 import { buildAd4mClient } from './util';
 import { showNotification } from '@mantine/notifications';
 
+import { appWindow } from '@tauri-apps/api/window'
+appWindow.listen('tauri://close-requested', ({ event, payload }) => {
```

Comment:
I'm not a react expert, should it be put into App component? FYI @fayeed

Reply:
Ideally, you would want to have it in a `useEffect`.